tracer/test/Datadog.Trace.ClrProfiler.IntegrationTests/NLogTests.cs

```diff
@@ -44,6 +45,39 @@ public static IEnumerable<object[]> GetTestData()
             }
         }
 
+        public static IEnumerable<object[]> GetContextTestData()
+        {
+            var minScopeContext = new Version("5.0.0");
+            var minMdlc = new Version("4.6.0");
+            var minMdc = new Version("4.0.0");
+            foreach (var item in PackageVersions.NLog)
+            {
+                var version = (string)item[0] == string.Empty ?
+                                  new Version("5.0.0") : // DEFAULT_SAMPLES
+                                  new Version((string)item[0]);
+
+                if (version >= minScopeContext)
+                {
+                    yield return item.Concat(false).Concat("ScopeContext");
+                    yield return item.Concat(true).Concat("ScopeContext");
+                }
+
+                if (version >= minMdlc)
+                {
+                    yield return item.Concat(false).Concat("Mdlc");
+                    yield return item.Concat(true).Concat("Mdlc");
+                }
+
+                if (version >= minMdc)
+                {
+                    yield return item.Concat(false).Concat("Mdc");
+                    yield return item.Concat(true).Concat("Mdc");
+                }
+
+                // No need to test without context properties, as that is thoroughly covered in other tests
```

Comment:
We should probably also test with `NDC` and `NDLC` - I don't think we have _any_ tests for those yet!

Also (non-blocking), I think you _could_ merge this "context" test with the existing "no context" test, given you already have to handle that case in the Program.cs. Just to reduce the duplication a bit 🙂

Reply:
Merged tests here, "I'll do that later" always ends up being never 😄 

Do you think it's worth testing NDC and NDLC? The tracer is not using or modifying them in any way, and AFAIK it never has and now with the introduction of `ScopeContext` never will.

Reply:
Thanks for doing that! Good question 🤔 My main reason for wanting to test it was knowing that it gets added to the context for direct log submission, but if we're not touching it, I don't know if there's any need tbh